modules/svg/gpx.js

```diff
@@ -139,7 +166,8 @@ export function svgGpx(projection, context, dispatch) {
     drawGpx.url = function(url) {
         d3.text(url, function(err, data) {
             if (!err) {
-                drawGpx.geojson(toGeoJSON.gpx(toDom(data)));
+                var extension = getExtension(url);
```

Comment:
FYI: this caused #3820 since url is just a string and getExtension() expects a File object. I fixed it by changing getExtension() to expect a string instead.